src/graphql.rs

```diff
@@ -329,6 +341,61 @@ where
     Ok((iter, cursor, size))
 }
 
+fn write_run_tcpdump(packets: &Vec<pk>) -> Result<String, anyhow::Error> {
+    let mut temp_file = tempfile()?;
+    let fd = temp_file.as_raw_fd();
+    let new_pcap = Capture::dead_with_precision(Linktype::ETHERNET, pcap::Precision::Nano)?;
+    let mut file = unsafe { new_pcap.savefile_raw_fd(fd)? };
+
+    for packet in packets {
+        let len = u32::try_from(packet.packet.len()).unwrap_or_default();
+        let header = PacketHeader {
+            ts: timeval {
+                tv_sec: packet.packet_timestamp / A_BILLION,
+                #[cfg(target_os = "macos")]
+                tv_usec: i32::try_from(packet.packet_timestamp & A_BILLION).unwrap_or_default(),
+                #[cfg(target_os = "linux")]
+                tv_usec: packet.packet_timestamp & A_BILLION,
+            },
+            caplen: len,
+            len,
+        };
+        let p = Packet {
+            header: &header,
+            data: &packet.packet,
+        };
+        file.write(&p);
+    }
+    let mut buf = Vec::new();
+    let _ = file.flush();
+    let _ = temp_file.seek(SeekFrom::Start(0));
+    let _ = temp_file.read_to_end(&mut buf)?;
+
+    let cmd = "tcpdump";
+    let args = ["-n", "-X", "-tttt", "-v", "-r", "-"];
+
+    let mut child = Command::new(cmd)
+        .env("PATH", "/usr/sbin:/usr/bin")
+        .args(args)
+        .stdin(Stdio::piped())
+        .stdout(Stdio::piped())
+        .spawn()?;
+
+    if let Some(mut child_stdin) = child.stdin.take() {
+        #[cfg(target_os = "macos")]
+        child_stdin.write_all(&[0, 0, 0, 0])?;
+        child_stdin.write_all(&buf)?;
+    } else {
+        return Err(anyhow!("failed to execute tcpdump"));
+    }
+
+    let output = child.wait_with_output()?;
+    if output.status.success() {
+        return Ok(String::from_utf8_lossy(&output.stdout).into_owned());
+    }
+
+    Err(anyhow!("failed to run tcpdump"))
```

Comment:
I think the final return looks better to be an `Ok` rather than `Err` even though I am not sure if doing so generates a clippy warning.

Reply:
Done